dotcom-rendering/src/web/components/Callout/Form.tsx

```diff
@@ -1,170 +1,200 @@
+	color: black;
+	${textSans.medium()};
+`;
 
 const formStyles = css`
 	display: flex;
 	flex-direction: column;
 	justify-content: space-between;
-	padding-left: 10px;
-	padding-right: 10px;
+	padding-left: ${space[2]}px;
+	padding-right: ${space[2]}px;
+`;
+
+const formFieldWrapperStyles = css`
+	display: flex;
+	flex-direction: column;
 `;
 
 const footerPaddingStyles = css`
 	display: flex;
 	flex-direction: row;
 	justify-content: space-between;
-	padding-bottom: 15px;
+	padding-bottom: ${space[4]}px;
 `;
 
-const errorMessagesStyles = css`
-	padding-bottom: 10px;
-	color: ${text.error};
-	${textSans.medium({ fontWeight: 'bold' })};
-`;
-
-const formFieldWrapperStyles = css`
-	display: flex;
-	flex-direction: column;
+const textStyles = css`
+	${textSans.medium()};
 `;
 
 type FormDataType = { [key in string]: any };
 
-type FormFieldProp = {
-	formField: CampaignFieldType;
-	formData: FormDataType;
-	setFormData: React.Dispatch<React.SetStateAction<FormDataType>>;
-};
-
-const FormField = ({ formField, formData, setFormData }: FormFieldProp) => {
-	switch (formField.type) {
-		case 'textarea':
-			return (
-				<>
-					<TextArea
-						formField={formField}
-						formData={formData}
-						setFormData={setFormData}
-					/>
-					<hr />
-				</>
-			);
-		case 'file':
-			return (
-				<>
-					<FileUpload
-						formField={formField}
-						formData={formData}
-						setFormData={setFormData}
-					/>
-					<hr />
-				</>
-			);
-		case 'select':
-			return (
-				<>
-					<Select
-						formField={formField}
-						formData={formData}
-						setFormData={setFormData}
-					/>
-					<hr />
-				</>
-			);
-		case 'checkbox':
-		case 'radio':
-			return (
-				<>
-					<MultiSelect
-						formField={formField}
-						formData={formData}
-						setFormData={setFormData}
-						multiple={formField.type === 'checkbox'}
-					/>
-					<hr />
-				</>
-			);
-		default: {
-			return (
-				<>
-					<TextInput
-						formField={formField}
-						formData={formData}
-						setFormData={setFormData}
-					/>
-					<hr />
-				</>
-			);
-		}
-	}
-};
-
 type FormProps = {
 	onSubmit: (formData: FormDataType) => void;
 	formFields: CampaignFieldType[];
+	format: ArticleFormat;
 	error?: string;
 };
 
-export const Form = ({ onSubmit, formFields, error }: FormProps) => {
-	const [twitterHandle, setTwitterHandle] = useState('');
-	const [formData, setFormData] = useState<{ [key in string]: any }>({});
+export const Form = ({ onSubmit, formFields, format, error }: FormProps) => {
+	const [formData, setFormData] = useState<FormDataType>({});
+	const [validationErrors, setValidationErrors] = useState<{
+		[key in string]: string;
+	}>({});
+
+	const setFieldInFormData = (
+		id: string,
+		data: string | string[] | undefined,
+	): void => {
+		const currentErrors = validationErrors;
+		currentErrors[id] = '';
+		setValidationErrors(currentErrors);
+
+		setFormData({
+			...formData,
+			[id]: data,
+		});
+	};
+
+	const validateForm = (): boolean => {
```

Comment:
Do you think this could become a function that receives parameters, instead of just `() => boolean`. 
That way , maybe we even move this function to a ts file to make this component a bit easier to read (this is just a suggestion)